tasks/main.yml

```diff
@@ -15,25 +9,89 @@
     - "{{ ansible_facts['distribution'] ~ '_' ~
           ansible_facts['distribution_version'] }}.yml"
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __vpn_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __vpn_vars_file is file
 
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __vpn_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-  notify: template handler to restart services
+- name: Enforce default encryption type as needed
+  set_fact:
+    vpn_connections: |
+      {% for tunnel in vpn_connections | d([]) %}
+      {%   set _ = tunnel.__setitem__(
+        "enc_type", tunnel.enc_type | d(vpn_enc_type)
+        ) %}
+      {% endfor %}
+      {{ vpn_connections | d([]) }}
+
+- name: copy psk values from a file
+  set_fact:
+    vpn_connections: |
+      {% for tunnel in vpn_connections | d([]) %}
+      {%   if tunnel.enc_type == "psk" %}
+      {%     if "shared_key_content" not in tunnel %}
+      {%       if "shared_key_src" in tunnel %}
+      {%         set _ = tunnel.__setitem__(
+        "shared_key_content", lookup('file', tunnel.shared_key_src)
+        ) %}
+      {%       endif %}
+      {%     endif %}
+      {%   endif %}
+      {% endfor %}
+      {{ vpn_connections | d([]) }}
+
+- name: generate random psks
+  shell: openssl rand -base64 48
+  register: psk
+  with_items: "{{ vpn_connections | d([]) }}"
+  run_once: true
```

Comment:
I think this task should be run on the controller host `localhost` using `delegate_to: localhost` or `local_action`: https://docs.ansible.com/ansible/latest/user_guide/playbooks_delegation.html#playbooks-delegation

Also, you'll have to mark this task with a `changed_when: false` https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#defining-changed so that ansible will never report that this task changed the state of the system.  This should clear up the ansible-lint error "[301] Commands should not change things if nothing needs doing"

Reply:
Also, you can use the `command` module here instead of `shell` which should clear up "[305] Use shell only when shell functionality is required" as well